web/client-api/src/main/java/io/deephaven/web/client/api/widget/plot/JsFigureFactory.java

```diff
@@ -40,66 +40,69 @@
         FigureDescriptor figureDescriptor = convertJsFigureDescriptor(descriptor);
         FetchObjectResponse response = new FetchObjectResponse();
         response.setData(figureDescriptor.serializeBinary());
-        // noinspection unchecked
-        Promise<JsTable>[] tableCopyPromises =
+        Promise<?>[] tableCopyPromises =
                 tables.map((table, index, all) -> table.copy(false)).asArray(new Promise[0]);
         return Promise.all(tableCopyPromises)
-                .then(tableCopies -> new JsFigure(
-                        c -> c.apply(null, response),
-                        (figure, descriptor1) -> {
-                            // We need to listen for disconnects and reconnects
-                            boolean[] isTableDisconnected = new boolean[tableCopies.length];
-                            ArrayList<RemoverFn> removerFns = new ArrayList<>(tableCopies.length * 3);
-
-                            for (int i = 0; i < tableCopies.length; i++) {
-                                final int tableIndex = i;
-                                // Tables are closed when the figure is closed, no need to remove listeners later
-                                removerFns.add(tableCopies[i].addEventListener(JsTable.EVENT_DISCONNECT, ignore -> {
-                                    isTableDisconnected[tableIndex] = true;
-                                    for (int j = 0; j < isTableDisconnected.length; j++) {
-                                        if (isTableDisconnected[j] && j != tableIndex) {
-                                            return;
+                .then(unknownTableCopies -> {
+                    JsTable[] tableCopies = Js.uncheckedCast(unknownTableCopies);
```

Comment:
If you change `tableCopies` to be of type `JsArray<JsTable>`, then you can do a plain Js.cast() instead of unchecked cast.